### DISCUSSION TITLE
Hype for next version 4.0

### DISCUSSION THREAD

**Discussion by trillox9** [jef/streetmerchant#2314] [General]
I'm in hype for the next release, the switch from puppeteer to cheerio will bring speed and less consuption of pc resources.
Do you know when update will available?

**Comment by 0000day** [comment]
Check the board: https://github.com/jef/streetmerchant/projects/4

**Reply by trillox9** [reply]
I've already seen, that's why I'm in hype for next release... I hope is released as soon as possible

**Reply by 0000day** [reply]
Well you can see the `to do's`. Nobody knows how long it will take. But it's not that much on the board.

**Reply by trillox9** [reply]
What do you think about switch from puppeteer to cheerio?

**Reply by jef** [reply]
I'm currently finishing up another project I'm working on and will have more time for streetmerchant in the near future. I will also be going through a cross country move soon, so it'll be intermittent. But there will be progress!

**Reply by jef** [reply]
That being said, the progress will be either followed on that board or via some branch.

**Reply by trillox9** [reply]
wow jef in person, thanks for your work... I hope as soon as possible but take your time man.
I think solution like puppeteer/selenium are too heavy for notification bot. 
The same things could be done with bs4/scrapy/cheerio with more lightness.
But that being said, I trust you, good work!

**Reply by jef** [reply]
Yep, refactoring to use cheerio is definitely in the works! I usually don't write in TypeScript or do frontend related work, so hindsight is definitely 20/20. This project outgrew itself within a couple weeks and haven't had much time to refactor a lot of it.

But! That being said, there will be time here very shortly and excited to see the outcome!

**Reply by trillox9** [reply]
I'm excited too! I looking forward to it!
Good work and thanks again

**Reply by trillox9** [reply]
@jef yo bro, sorry for disturbing you.
Any news for 4.0 release? :)

**Reply by jef** [reply]
Hey @trillox9, unfortunately not at this time. I have been pretty busy with work and I'm moving cross country, but I hope to see some changes in the near future.

**Reply by trillox9** [reply]
Thanks for your reply @jef, let's hope soon :)

**Reply by Aragorn450** [reply]
Hey @jef, why the change to Cheerio? It seems to me that it'd cause issues with the anti-bot protection that is on a number of sites since it doesn't parse JavaScript and such... Cheerio is great for what it's built for, but it lacks in a LOT of ways for what is required here.

**Reply by jef** [reply]
I don't believe Cheerio will be the solution. I think something like jsdom may be a better solution. I haven't fully thought this one out yet, but perhaps something like this.

**Reply by Aragorn450** [reply]
But why stop using a "browser" at all? At least with an actual browser that's controlled, it looks more like a regular user. Using these other things mean it's not necessarily behaving like a browser and therefore will be more likely to look like what it is; a bot scraping service.

What would you be trying to accomplish by switching at all?

**Reply by trillox9** [reply]
Browser Is heavy and useless for check only availability... and is recognized as bot anyway.
@jef i think the best solution is python with bs4 or scaper, could you look at this?

**Reply by Aragorn450** [reply]
I can pretty much guarantee that anything other than a browser will be identified as a bot even faster than Puppeteer is. Yes, it's heavy, although no heavier than Chrome itself.

And it has a lot of good APIs for finding content and interacting with it, which means you can do things like add to cart which wouldn't be possible with a simple scraper.

If you want to try a Python-based scraper, feel free to try that. But please don't ask @jef to do that for you...

**Reply by trillox9** [reply]
Jef is already directed toward that end.
Is the best way for speed and resource savings.
You can simply not update if you prefer puppeteer, 3.5.0 works well...

**Reply by jef** [reply]
I haven't gone through all the options and perhaps puppeteer will be our best bet! Optimizing that may be something I'd rather work on.

That being said, I think changing the backend for getting these links are lower priority then having a dynamic (internal) API that updates links/people can contribute to; rather than what we have currently (static models).
